src/detect-engine-port.c

```diff
@@ -1413,21 +1414,25 @@ DetectPort *PortParse(const char *str)
     }
 
     /* see if the address is an ipv4 or ipv6 address */
-    if ((port2 = strchr(port, ':')) != NULL)  {
+    if ((port2 = strchr(port, ':')) != NULL) {
         /* 80:81 range format */
         port2[0] = '\0';
         port2++;
 
-        if(DetectPortIsValidRange(port))
-            dp->port = atoi(port);
-        else
-            goto error;
+        if (strcmp(port, "") != 0) {
```

Comment:
Why do we consider empty string means port 0 and not failure ?

Reply:
hmm I had thought that since port2's value in case it is empty, it is reset to 65535 then this should be reset to 0 since it is range of ports we're working on. So if we have `:5000`, it should probably be `0:5000`.
If this is incorrect which I believe it is from your comment, I would like to learn why.

Reply:
Ok thanks, I did not know this was in a range.

Reply:
So, should it be 0 or 0xFFFF ?

Reply:
I think it should be 0 since strcmp returns 0 in case strings are equal. In this case, if start port is empty then strcmp would return 0.

Reply:
Looks ok to me, thanks